src/attr/_make.py

```diff
@@ -2724,6 +2724,25 @@ def __init__(self, converter, *, takes_self=False, takes_field=False):
             converter
         ).get_first_param_type()
 
+        self.__call__ = {
+            (False, False): self._takes_only_value,
+            (True, False): self._takes_instance,
+            (False, True): self._takes_field,
+            (True, True): self._takes_both,
+        }[self.takes_self, self.takes_field]
+
+    def _takes_only_value(self, value, instance, field):
+        return self.converter(value)
+
+    def _takes_instance(self, value, instance, field):
+        return self.converter(value, instance)
+
+    def _takes_field(self, value, instance, field):
+        return self.converter(value, field)
+
+    def _takes_both(self, value, instance, field):
+        return self.converter(value, instance, field)
+
```

Comment:
@Tinche how about now? here, the dict even makes a bit of sense ;) bummer we need `self`, otherwise I would make it a class var too